src/main/java/com/faforever/client/preferences/ForgedAlliancePrefs.java

```diff
@@ -63,13 +70,26 @@ public ForgedAlliancePrefs() {
     port = new SimpleIntegerProperty(6112);
     path = new SimpleObjectProperty<>();
     installationPath = new SimpleObjectProperty<>();
-    customMapsDirectory = new SimpleObjectProperty<>(GPG_FA_PATH.resolve("Maps"));
     officialMapsDirectory = new SimpleObjectProperty<>(STEAM_FA_PATH.resolve("Maps"));
-    modsDirectory = new SimpleObjectProperty<>(GPG_FA_PATH.resolve("Mods"));
+    vaultBaseDirectory = new SimpleObjectProperty<>(GPG_FA_PATH);
+    customMapsDirectory = new SimpleObjectProperty<>(getVaultBaseDirectory().resolve("Maps"));
+    modsDirectory = new SimpleObjectProperty<>(getVaultBaseDirectory().resolve("Mods"));
     preferencesFile = new SimpleObjectProperty<>(LOCAL_FA_DATA_PATH.resolve("Game.prefs"));
     autoDownloadMaps = new SimpleBooleanProperty(true);
     executableDecorator = new SimpleStringProperty("\"%s\"");
     executionDirectory = new SimpleObjectProperty<>();
+    vaultCheckDone = new SimpleBooleanProperty(false);
+    bindVaultPath();
+  }
+
+  /**
+   * Needs to be called after gson deserialization.
+   */
+  protected void bindVaultPath() {
+    customMapsDirectory.unbind();
```

Comment:
The problem here is gson recreating the object with new customMapsDirectory  etc. I wanted to stop gson from deserializing it but failed, if i do so other strange exceptions happen maybe somebody can have a look at it... https://stackoverflow.com/questions/4802887/gson-how-to-exclude-specific-fields-from-serialization-without-annotations